test/specs/reference/tests/10.update.js

```diff
@@ -110,6 +113,138 @@ exports.execute = function (options) {
+            var reference, column,
+                columnName = "uri",
+                tableName = "file_update_table",
+                sortBy = "key",
+                baseUri = options.url + "/catalog/" + catalogId + "/entity/" + schemaName + ':' + tableName + "/key=1";
+
+            var file = {
+                name: "testfile500kb.png",
+                size: 512000,
+                displaySize: "500KB",
+                type: "image/png",
+                hash: "4b178700e5f3b15ce799f2c6c1465741",
+                hash_64: "SxeHAOXzsVznmfLGwUZXQQ=="
+            }
+
+            beforeAll(function (done) {
+                var filePath = "test/specs/reference/files/" + file.name;
+
+                exec("perl -e 'print \"\1\" x " + file.size + "' > " + filePath);
+
+	        	file.file = new File(filePath);
+
+                options.ermRest.resolve(baseUri, { cid: "test" }).then(function (response) {
+                    reference = response;
+                    reference = reference.contextualize.entryEdit;
+
+                    column = reference.columns.find(function(c) { return c.name == columnName;  });
+
+                	if (!column) {
+                		console.log("Unable to find column " + columnName);
+                		done.fail();
+                		return;
+                	}
+                    done();
+                }, function (err) {
+                    console.dir(err);
+                    done.fail();
+                });
+            });
+
+            it("the metadata for the asset should be updated.", function (done) {
+                var tuples, tuple, fileUrl, updateData = {},
+                    baseUrl = options.url.replace("/ermrest", "");
+
+                var validRow = { key: 1, uri : { md5_hex: file.hash } };
+
+                var uploadObj = new options.ermRest.Upload(file.file, {
+                    column: column,
+                    reference: reference,
+                    chunkSize: 5 * 1024 * 1024
+                });
+
+                expect(uploadObj instanceof options.ermRest.Upload).toBe(true);
+
+                reference.read(1).then(function(response) {
+                    tuples = response.tuples;
+                    tuple = tuples[0];
+
+                    return uploadObj.calculateChecksum(validRow);
+                }).then(function(url) {
+                    expect(url).toBe("/hatrac/" + validRow.key + "/" + file.hash, "File generated url is not the same");
+
+                    expect(validRow.filename).toBe(file.name);
+                    expect(validRow.bytes).toBe(file.size);
+                    expect(validRow.File_MD5).toBe(file.hash);
+
+                    return uploadObj.createUploadJob();
+                }).then(function(url) {
+                    expect(url).toBeDefined("Chunk url not returned");
+
+                    return uploadObj.fileExists();
+                }).then(function() {
+                    expect(true).toBe(true);
```

Comment:
So how I started writing this test was by mimicking the tests that @chiragsanghvi wrote specifically for file upload. I wasn't sure what was crucial for the upload task and what wasn't.

This actually looks like a sanity check and can probably be removed under the assumption that the file exists because another spec is testing for that.